crates/anstyle-svg/src/lib.rs

```diff
@@ -92,18 +100,8 @@ impl Term {
 
         let mut styled = adapter::AnsiBytes::new();
         let mut elements = styled.extract_next(ansi.as_bytes()).collect::<Vec<_>>();
-        let mut effects_in_use = anstyle::Effects::new();
-        for element in &mut elements {
-            let style = &mut element.style;
-            // Pre-process INVERT to make fg/bg calculations easier
-            if style.get_effects().contains(anstyle::Effects::INVERT) {
-                *style = style
-                    .fg_color(Some(style.get_bg_color().unwrap_or(self.bg_color)))
-                    .bg_color(Some(style.get_fg_color().unwrap_or(self.fg_color)))
-                    .effects(style.get_effects().remove(anstyle::Effects::INVERT));
-            }
-            effects_in_use |= style.get_effects();
-        }
+        preprocess_invert_style(&mut elements, self.bg_color, self.fg_color);
```

Comment:
Could you split these refactors out into a commit before this one so I can better focus on the intent of each change?

Reply:
Sure, done.